components/framework-axp/src/main/java/framework/configuration/impl/ConfigFileReaderServiceImpl.java

```diff
@@ -0,0 +1,86 @@
+package framework.configuration.impl;
+
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import framework.cache.AXPCacheBuilder;
+import framework.cache.IAXPCacheLoader;
+import framework.cache.IAXPLoadingCache;
+import framework.configuration.ConfigFile;
+import framework.configuration.ConfigFileReaderStrategy;
+import framework.configuration.IConfigFileReaderService;
+import framework.configuration.IConfigFileReaderStrategy;
+
+/**
+ * Implementation for the {@link IConfigFileReaderService}
+ * This file reader class will help to read configuration property files with a caching support.
+ */
+public class ConfigFileReaderServiceImpl implements IConfigFileReaderService
```

Comment:
It would be useful to have a ConfigReaderService which have multiple implementations. One implementation would have AXPCache for storing properties, while a different implementation may use a simple HashMap.